utils/python/CIME/XML/generic_xml.py

```diff
@@ -186,10 +187,19 @@ def get_resolved_value(self, raw_value):
             logger.debug("find: %s" % var)
             ref = self.get_value(var)
             if ref is not None:
-                logger.debug("resolve: %s" % ref)
-                item_data = item_data.replace(m.group(), str(self.get_resolved_value(ref)))
+                logger.debug("resolve: " + str(ref))
+                item_data = item_data.replace(m.group(), self.get_resolved_value(str(ref)))
+            elif var == "CIMEROOT":
```

Comment:
These two variables cause a lot of errors - handling them directly just makes life easier.